frame/support/src/lib.rs

```diff
@@ -1595,33 +1595,22 @@ pub mod pallet_prelude {
 /// 	// WARNING: Each parameter used in functions must implement: Clone, Debug, Eq, PartialEq,
 /// 	// Codec.
 /// 	//
-/// 	// The macro checks:
-/// 	// * pallet is `Pallet<T>` or `Pallet<T, I>`
-/// 	// * trait is `Call`
-/// 	// * each dispatchable functions first argument is `origin: OriginFor<T>` (OriginFor is
-/// 	//   imported from frame_system.
-/// 	//
-/// 	// The macro parse `#[pallet::compact]` attributes, function parameter with this attribute
-/// 	// will be encoded/decoded using compact codec in implementation of codec for the enum
-/// 	// `Call`.
-/// 	//
-/// 	// The macro generate the enum `Call` with a variant for each dispatchable and implements
-/// 	// codec, Eq, PartialEq, Clone and Debug.
+/// 	// The macro parses `#[pallet::compact]` attributes on function arguments, and use it to
+/// 	// implement `Call` encoding/decoding.
 /// 	#[pallet::call]
 /// 	impl<T: Config> Pallet<T> {
 /// 		/// Doc comment put in metadata
 /// 		#[pallet::weight(0)] // Defines weight for call (function parameters are in scope)
 /// 		fn toto(
 /// 			origin: OriginFor<T>,
-/// 			#[pallet::compact] _foo: u32
+/// 			#[pallet::compact] _foo: u32,
```

Comment:
dq: how is `#[pallet::compact]` different from `#[scale(compact)]`?

Reply:
I think the intended semantics are the same. I imagine it's easier for the pallet macro to parse this way.

Reply:
yes I used `pallet` keyword for every attribute which are parsed (and removed) by pallet macro.
This attribute will be used to add the `#[scale(compact)]` on the field when generating the call enum.